sys/kern/vm_map.c

```diff
@@ -220,28 +220,39 @@ void vm_map_delete(vm_map_t *map) {
   pool_free(P_VM_MAP, map);
 }
 
-/* TODO(fzdo): allow for changing protection bits of parts of entries */
-/* XXX: This function allows for setting protection bits fo existing entries
- * only. It can't change protection of part of entry (currently we don't need to
- * set protection of part of entry). */
+static inline bool range_intersects_map_entry(vm_map_entry_t *ent,
+                                              vaddr_t start, vaddr_t end) {
+  return vm_map_entry_end(ent) > start && vm_map_entry_start(ent) < end;
+}
+
 void vm_map_protect(vm_map_t *map, vaddr_t start, vaddr_t end, vm_prot_t prot) {
```

Comment:
Please look through respective `mprotect` manual pages:
 * [OpenGroup](https://pubs.opengroup.org/onlinepubs/9699919799/functions/mprotect.html),
 * [NetBSD](https://man.netbsd.org/mprotect.2),
 * [Linux](https://man7.org/linux/man-pages/man2/mprotect.2.html)
and decide if some of the errors should returned by `vm_map_protect` function, otherwise return `0`.

Reply:
I have found, in OpenGroup manual, that if some addresses withing given range are not mapped, then we should return `ENOMEM`. I have fixed that and added few checks in tests for that case.